src/main/scala/com/mongodb/spark/rdd/partitioner/BsonValueOrdering.scala

```diff
@@ -175,6 +172,50 @@ trait BsonValueOrdering extends Ordering[BsonValue] {
     }
   }
 
+  private def compareStrings(x: BsonValue, y: BsonValue): Int = {
+    val (xString, yString) = (x.getBsonType, y.getBsonType) match {
+      case (BsonType.STRING, BsonType.STRING) => (x.asString.getValue, y.asString.getValue)
+      case (BsonType.SYMBOL, BsonType.STRING) => (x.asSymbol.getSymbol, y.asString.getValue)
+      case (BsonType.STRING, BsonType.SYMBOL) => (x.asString.getValue, y.asSymbol.getSymbol)
+      case (BsonType.SYMBOL, BsonType.SYMBOL) => (x.asSymbol.getSymbol, y.asSymbol.getSymbol)
+      case _ => throw new UnsupportedOperationException(s"Cannot compare $x with $y. Not string compatible types.")
+    }
+    compareBytes(xString.getBytes("utf-8") zip yString.getBytes("utf-8"))
+  }
+
+  private def compareNumbers(x: BsonNumber, y: BsonNumber): Int = {
+    (x.getBsonType, y.getBsonType) match {
+      case (BsonType.INT64, BsonType.DOUBLE) => compareLongToDouble(x.longValue, y.doubleValue)
+      case (BsonType.DOUBLE, BsonType.INT64) => - compareLongToDouble(y.longValue, x.doubleValue)
+      case _ => x.doubleValue.compareTo(y.doubleValue)
+    }
+  }
+
+  private def compareLongToDouble(x: Long, y: Double): Int = {
+    val END_OF_PRECISE_DOUBLES: Double = 1 << 53 // positive 2**63
+    val BOUND_OF_LONG_RANGE: Double = Long.MaxValue.toDouble
```

Comment:
Ah that explains my errors: In Java it was `BOUND_OF_LONG_RANGE = -(double) Long.MAX_VALUE;` not MIN_VALUE.